### ISSUE TITLE
Repeated deployment crashing on disk blob already existing

### ISSUE THREAD

**Issue by guydav** [wbuchwalter/blog-posts#4]
Hi William,

As I mentioned in my comment to the post, I repeatedly run into an error with deploying. I'm on a fully updated pull of your k8s-gpu branch, in a freshly setup devenv. 

This is the json I'm working from, where I replaced key data and ids with placeholders:
```json
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorVersion": "1.6.2"
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "KubernetesGPUTest",
      "vmSize": "Standard_D2_v2"
    },
    "agentPoolProfiles": [
      {
        "name": "gpuagent",
        "count": 1,
        "vmSize": "Standard_NV12",
        "availabilityProfile": "AvailabilitySet"
      }
    ],
    "linuxProfile": {
      "adminUsername": "admin",
      "ssh": {
        "publicKeys": [
          {
            "keyData": "SOME RSA KEY DATA"
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "servicePrincipalClientID": "SOME UUID",
      "servicePrincipalClientSecret": "SOME SECRET"
    }
  }
}
```

When I run the following execution, it fails:
```
az group deployment create --template-file azuredeploy.json --parameters @azuredeploy.parameters.json --resource-group GPUs
```

With the following error:
```json
{
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "TargetDiskBlobAlreadyExists",
        "message": "Blob https://006o454eidd57rsagnt0.blob.core.windows.net/osdisk/k8s-gpuagent-61084489-0-osdisk.vhd already exists. Please provide a different blob URI as target for disk 'k8s-gpuagent-61084489-0-osdisk'."
      }
    ]
  }
}  
```

The resource group GPUs exists already, and has other resources in it. Might that be an issue? Why would it? This failed deployment creates several resources with the k8s tag in its name, which I manually deleted between failed deployments, so I don't think it's the remnants of previous failed deployments that are causing it to fail.

This is the initial deployment, so before autoscaling, any kubernetes cluster configuration, or anything fancier is going on. 

Any advice?

Thanks!

**Comment by wbuchwalter** [comment]
> This failed deployment creates several resources with the k8s tag in its name, which I manually deleted between failed deployment

The OS disks are stored in a storage account whose name does not start with `k8s`, in your case it should be `006o454eidd57rsagnt0`. Can you make sure this storage account is deleted before deploying?

Generally it's best to deploy in an empty resource group to avoid this kind of conflicts though.

**Comment by guydav** [comment]
That makes a lot of sense. I should have picked that up from the error message. I cleaned up the storage accounts as well. I guess it's tricky since the indicative names appear in the containers, which are only searchable from the storage account itself, not the overall resource search.

I vaguely recall why I tried to deploy into an existing resource group, but that reason is irrelevant at this point. If I end up needing to clean up this deployment - does deleting the resource group delete everything underneath it? Or will I have to go do more manual cleanup afterwards?

Oh, and owing to the fact I deployed into a new group, and cleaned up the previous storage accounts, the deployment went smoothly.
